agents-for-bedrock/features-examples/08-create-agent-with-guardrails/08-create-agent-with-guardrails.ipynb

```diff
@@ -0,0 +1,1956 @@
+   "id": "a8e0734a-82ce-4524-b02f-446ff80c54d4",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "#### Create the required permissions\n",
+    "Now let's also create the lambda role and its required policies. For this case, we need the lambda to be able to access DynamoDB, that is why we also create a DynamoDB policy and attach to our Lambda. To do so, we will use the support function `create_lambda_role`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71b184da-0254-4ff7-a19f-2053a86e0582",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "#### Create the function\n",
+    "\n",
+    "Now that we have the Lambda function code and its execution role, let's package it into a Zip file and create the Lambda resources"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "855bf5e0-0345-466f-96dd-4fd436455539",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### 4.1 - 4.3 Create the Agent\n",
+    "Now that we have created the Knowledge Base and the Lambda function to execute the tasks for our agent, let's start creating our Agent.\n",
+    "\n",
+    "First need to create the agent policies that allow bedrock model invocation and Knowledge Base query and the agent IAM role with the policy associated to it. We will allow this agent to invoke the Claude Sonnet model. Then we would need to actually create the agent while associating that role with the agent using (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-agent/client/create_agent.html) api from boto3. It requires an agent name, underline foundation model and instruction. You can also provide an agent description.\n",
+    "\n",
+    "Here we use the [`create_agent`] function from agent.py to both create the IAM role and the Agent itself  \n",
+    "\n",
+    "Note that the agent created is not yet prepared. We will focus on preparing the agent and then using it to invoke actions and use other APIs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1021cade-ceea-498b-900d-fb1c5a7916e8",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "#Command to create agent using boto3\n",
+    "'''\n",
+    "response = bedrock_agent_client.create_agent(\n",
+    "    agentName=agent_name,\n",
+    "    agentResourceRoleArn=agent_role['Role']['Arn'],\n",
+    "    description=agent_description,\n",
+    "    idleSessionTTLInSeconds=1800,\n",
+    "    foundationModel=agent_foundation_model,\n",
+    "    instruction=agent_instruction,\n",
+    ")\n",
+    "response\n",
+    "'''\n",
+    "kb_arn =  f\"arn:aws:bedrock:us-west-2:478242907172:knowledge-base/{kb_id}\"\n",
+    "agent_id = agents.create_agent(agent_name, agent_description, agent_instruction, [agent_foundation_model], kb_arns=[kb_arn])\n",
```

Comment:
agents is not defined.
you need to add
`agents = AgentsForAmazonBedrock()` before for it to work